docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
+4. Explicitly run the last computation with `Async.Start`
 
-* `let!` binds the result of an async expression (which runs on another context).
-* `use!` works just like `let!`, but disposes its bound resources when it goes out of scope.
-* `do!` will await an async workflow which doesn’t return anything.
-* `return` simply returns a result from an async expression.
-* `return!` executes another async workflow and returns its return value as a result.
+When this program runs, `countFileBytes` runs in parallel for each command-line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
 
-Additionally, normal `let`, `use`, and `do` keywords can be used alongside the async versions just as they would in a normal function.
+## Sequencing asynchronous computations
 
-## How to start Async Code in F\#
+Because `Async<'T>` is a specification of work rather than an already-running task, you can perform more intricate transformations easily. Here is an example where you can sequence a set of computations so they execute one after another.
 
-As mentioned earlier, async code is a specification of work to be done in another context which needs to be explicitly started. Here are two primary ways to accomplish this:
+```fsharp
+let countFileBytes path =
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
+    }
+
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Sequential
+    |> Async.RunSynchronously
+    |> ignore
+```
+
+This will execute `countFileBytes` for each item in `argv` in the order of the elements of `argv` rather than executing them in parallel.
 
-1. `Async.RunSynchronously` will start an async workflow on another thread and await its result.
+## Key async module functions
 
-    ```fsharp
-    open System
-    open System.Net
+When you write async code in F# you'll usually interact with a framework that handles scheduling of computations for you. However, this is not always the case, so it is good to learn the various starting functions to schedule asynchronous work.
 
-    let fetchHtmlAsync url =
-        async {
-            let uri = Uri(url)
-            use webClient = new WebClient()
-            let! html = webClient.AsyncDownloadString(uri)
-            return html
-        }
+Because F# asynchronous computations are a _specification_ of work rather than a representation of work that is already executing, they must be explicitly started with a starting function. There are many [Async starting functions](https://msdn.microsoft.com/library/ee370232.aspx) that are helpful in different contexts.The following section describes some of the more common starting functions.
```

Comment:
```suggestion
Because F# asynchronous computations are a *specification* of work rather than a representation of work that is already executing, they must be explicitly started with a starting function. There are many [Async starting functions](https://msdn.microsoft.com/library/ee370232.aspx) that are helpful in different contexts.The following section describes some of the more common starting functions.
```